src/Data/Song.cpp

```diff
@@ -40,23 +41,30 @@ namespace SongDetailsCache {
 
     float Song::minNJS() const noexcept { return min([](const auto& diff){ return diff.njs; }); }
     float Song::maxNJS() const noexcept { return max([](const auto& diff){ return diff.njs; }); }
-    float Song::minStar() const noexcept { 
-        if (rankedStatus != RankedStatus::Ranked) return 0.0f;
-        return min([](const auto& diff){ return diff.ranked() ? diff.stars : std::numeric_limits<float>::max(); }); 
+    float Song::minStarBL() const noexcept { 
+        if (rankedStates != RankedStates::BeatleaderRanked) return 0.0f;
+        return min([](const auto& diff){ return diff.rankedSS() ? diff.starsBL : std::numeric_limits<float>::max(); }); 
```

Comment:
this check is wrong as it uses rankedSS to check for BL rankedness

Reply:
also please use the `hasFlags` method for the if checks, instead of `rankedStates != RankedStates::BeatleaderRanked`